src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1125,6 +1125,63 @@ protected static void VerifyOperationTreeForTest<TSyntaxNode>(CSharpCompilation
             additionalOperationTreeVerifier?.Invoke(actualOperation, compilation, syntaxNode);
         }
 
+        protected static void VerifyFlowGraphForTest<TSyntaxNode>(CSharpCompilation compilation, string expectedFlowGraph)
+            where TSyntaxNode : SyntaxNode
+        {
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+            SyntaxNode syntaxNode = GetSyntaxNodeOfTypeForBinding<TSyntaxNode>(GetSyntaxNodeList(tree));
+
+            ImmutableArray<Operations.BasicBlock> graph = model.GetControlFlowGraph((Operations.IBlockOperation)model.GetOperation(syntaxNode));
+            var map = new Dictionary<Operations.BasicBlock, int>();
+
+            for (int i = 0; i < graph.Length; i++)
+            {
+                map.Add(graph[i], i);
+            }
+
+            var stringBuilder = PooledObjects.PooledStringBuilder.GetInstance();
+
+            for (int i = 0; i < graph.Length; i++)
+            {
+                var block = graph[i];
+                stringBuilder.Builder.AppendLine($"Block[{i}] - {block.Kind}");
+
+                var predecessors = block.Predecessors;
+
+                if (!predecessors.IsEmpty)
+                {
+                    stringBuilder.Builder.AppendLine($"    Predecessors ({predecessors.Count})");
+                    foreach (int j in predecessors.Select(b => map[b]).OrderBy(ii => ii))
+                    {
+                        stringBuilder.Builder.AppendLine($"        [{j}]");
+                    }
+
+                }
+
+                var statements = block.Statements;
+                stringBuilder.Builder.AppendLine($"    Statements ({statements.Length})");
+                foreach (var statement in statements)
+                {
+                    stringBuilder.Builder.AppendLine(OperationTreeVerifier.GetOperationTree(compilation, statement, initialIndent: 8));
+                }
+
+                if (block.Conditional.Condition != null)
+                {
+                    stringBuilder.Builder.AppendLine($"    Jump if {block.Conditional.JumpIfTrue} to Block[{(map.TryGetValue(block.Conditional.Destination, out var index) ? index : -1)}]");
```

Comment:
The case at the end of this line where the destination index gets defaulted to `-1` means the graph is busted. 
No tests should ever print this `-1`. Consider asserting instead.